server/init.lua

```diff
@@ -23,17 +26,53 @@ event.listen("component_added", updateMachineList)
 
 local statuses = {multiblocks = {}, power = {}}
 
+local function updateMachineStatuses(_evName, _localAddress, sender, port, _distance, ...)
+    local args = {...}
+    if port == portNumber and args[1] == "local_multiblock_statuses" then
+        for address, status in pairs(serialization.unserialize(args[2])) do
+            statuses.multiblocks[address] = status
+        end
+    end
+end
+event.listen("modem_message", updateMachineStatuses)
+
+local function updatePowerStatus(_evName, _localAddress, sender, port, _distance, ...)
+    local args = {...}
+    if port == portNumber and args[1] == "local_power_status" then
+        statuses.powerStatus = serialization.unserialize(args[2])
+    end
+end
+event.listen("modem_message", updatePowerStatus)
+
 function server.configure(x, y)
     -- TODO: Code for GUI configuration of machines:
     ---- Machine renaming
     ---- Machine widgets layout
 end
 
+function server.load()
+    -- TODO: Ping other servers for their statuses on boot up
+    ---- Check if there's no response and determine if this is the main server
+end
+
 function server.update()
+    local shouldBroadcastStatuses = false
     for address, name in pairs(machineAddresses) do
-        statuses.multiblocks[address] = getMultiblockStatus(address, name)
+        local multiblockStatus = getMultiblockStatus(address, name)
+        shouldBroadcastStatuses = statuses.multiblocks[address].state ~= multiblockStatus.state
+        statuses.multiblocks[address] = multiblockStatus
+    end
+    if shouldBroadcastStatuses then
+        modem.broadcast(portNumber, "local_multiblock_statuses", serialization.serialize(statuses.multiblocks))
```

Comment:
Should this gather a table of all the statuses that changed and only send those to avoid network congestion?

No need to send the data from 50 machines if one got a maintenance problem.

Reply:
🤔

Reply:
Another reasoning: The max packet size for one message is 8192 bytes. I would guess one status is around 200-300 bytes long when serialized, which means sending a full server of 50 machines will fail if you try to send the whole list.

Also, only problem amount and state should be sent, along with the address.

Reply:
Only that?
I thought the main server would be able to display all sorts of informations

Reply:
Alright

Reply:
> Only that?
> I thought the main server would be able to display all sorts of informations

It _can_, but network events should only happen around once a minute or so ideally, and it's not very engaging watching a progress bar that updates once a minute.